lib/Group_Proxy.php

```diff
@@ -117,10 +127,15 @@ public function inGroup($uid, $gid) {
 	public function getUserGroups($uid) {
 		$groups = array();
 
-		foreach($this->backends as $backend) {
-			$backendGroups = $backend->getUserGroups($uid);
-			if (is_array($backendGroups)) {
-				$groups = array_merge($groups, $backendGroups);
+		foreach($this->backends as $configPrefix => $backend) {
```

Comment:
There is a risk here that a process calls `getUserGroups()` and sees a user suddendly isn't in a specific group any more. This means that if that user received shares over one of the "dead" groups, the share will disappear from the filesystem and sync clients will start deleting it locally.

We need to make sure the shares are still there but throw `StorageNotAvailableException`. Due to the twisted complexity of the architecture I don't think this is easily doable.